request.go

```diff
@@ -107,10 +107,13 @@ func (r *callRequest) Reply() error {
 			r.err,
 			r.res,
 		}
-		err = r.writer.Encode(v)
-		if err != nil {
-			r.log.Warning("Reply error for %d: %s", r.seqno, err.Error())
-		}
+		errCh := r.writer.Encode(v)
```

Comment:
Do something similar here with doing the Encode first, and selecting on ctx.Done and errCh